src/golomb.h

```diff
@@ -96,40 +97,39 @@ class encoder
     int         encoded_count;
 
 public:
-    encoder( OutputIt output, size_t k = 0u ) noexcept
-        : output( output )
-        , k( k )
-        , base( 1u << k )
-        , zeros_threshold( 1 + k )
+    encoder( OutputIt output_in, size_t k_in = 0u ) noexcept
+        : output( output_in )
+        , k( k_in)
+        , base( 1u << k_in)
+        , zeros_threshold( 1 + k_in)
         , encoded( static_cast< OutputDataT>( 0u ) )
         , encoded_count( 0 )
     {}
 
-    template< typename InputValueT >
-    requires std::integral< InputValueT >
+    template< std::integral InputValueT >
     OutputIt push( InputValueT x )
     {
         using UnsignedInputValueT = typename std::make_unsigned< InputValueT >::type;
         using CommonT             = typename std::common_type< UnsignedInputValueT, OutputDataT >::type;
 
-        constexpr auto input_digits  = std::numeric_limits< UnsignedInputValueT >::digits;
-        constexpr auto input_mask    = std::numeric_limits< UnsignedInputValueT >::max();
+        static constexpr auto input_digits  = std::numeric_limits< UnsignedInputValueT >::digits;
+        static constexpr auto input_mask    = std::numeric_limits< UnsignedInputValueT >::max();
 
-        const auto overflow_threshold = std::numeric_limits< UnsignedInputValueT >::max() - ( static_cast< OutputDataT >( base ) );
+        const auto overflow_threshold = input_mask - ( static_cast< OutputDataT >( base ) );
         const auto max_zeros          = input_digits - k;
 
         const auto u         = detail::to_unsigned( x );
         const bool overflow  = u > overflow_threshold;
         const auto value     = static_cast< UnsignedInputValueT >( u + base );
         const int  bit_width = static_cast< int >( std::bit_width( value ) ); // cast is a workaround for unresolved defect report in GCC/libc++
         const int  width     = overflow ? input_digits : bit_width;
-        const int  zeros     = overflow ? max_zeros : width - zeros_threshold;
+        const int  zeros     = overflow ? max_zeros : (width - zeros_threshold);
```

Comment:
The braces are not needed here.
I'm a bit reluctant to use braces 'just to be sure'. A temporary value is more preferable but i think even that is not needed here.

Reply:
Personal preference, I like to clarify this kind of stuff for future me, when debugging these steps are often uncertain. Was the purpose a turnary and then substracting zeros_threshold from the turnary result, or is one of the turnary results with - zeros_threshold?
In this case, it is the latter and the brackets are not required.